lxd/db/openfga/openfga.go

```diff
@@ -248,10 +345,93 @@ func (o *openfgaStore) ReadUsersetTuples(ctx context.Context, store string, filt
 		return nil, fmt.Errorf("ReadUsersetTuples: Failed to parse entity URL %q: %w", entityURL, err)
 	}
 
-	var groupNames []string
+	// Get cache from context. If it is not present, we'll fall back to calling the database directly.
+	cache, err := request.GetCtxValue[*RequestCache](ctx, request.CtxOpenFGARequestCache)
```

Comment:
why do we need this section, if the cache is not in the context we have a bug right? Why would we mask over that here?

Reply:
I asked @gabrielmougard to do this in the original PR; which performed caching slightly differently, but the reasoning is the same.

The reason is that for long running requests we may not want to use the cache. For example, currently have an open issue to improve authorization for the events API. I anticipate that this will check `can_view` on a per-resource basis. If we use this cache for this kind of request, we'll have a snapshot of permissions at the time of the original request, but this may change while the web socket is still connected.

So this logic is in anticipation of setting the value of the request cache to nil in the request context, in the edge cases where we don't want to use a cache.